deta/deta.go

```diff
@@ -0,0 +1,92 @@
+/*
+Package deta is the official Deta SDK for Go
+
+Example:
+	package main
+
+	import (
+		"fmt"
+
+		"github.com/deta/deta-go/deta"
+		"github.com/deta/deta-go/service/base"
+	)
+
+	type User struct {
+		Key      string `json:"key"` // json struct tag key to denote the key
+		Username string `json:"username"`
+		Email    string `json:"email"`
+	}
+
+	func main() {
+		d, err := deta.New(deta.WithProjectKey("project_key"))
+		if err != nil {
+			fmt.Println("failed to init new Deta instance:", err)
+			return
+		}
+
+		db := base.New(d, "users")
+
+		u := &User{
+			Key:      "abasd",
+			Username: "jimmy",
+			Email:    "jimmy@deta.sh",
+		}
+		key, err := db.Put(u)
+		if err != nil {
+			fmt.Println("failed to put item:", err)
+			return
+		}
+		fmt.Println("successfully put item with key", key)
+	}
+
+More examples on https://docs.deta.sh/docs/base/sdk
+*/
+package deta
+
+import (
+	"errors"
+	"os"
+	"strings"
+)
+
+var (
+	// ErrBadProjectKey bad project key
+	ErrBadProjectKey = errors.New("bad project key")
```

Comment:
can you please collect all the exported errors also in `internal/client` and put it in a file called `deta/errors.go` and export them from there?

Reply:
can you please also put this in `deta/errors.go`